README.md

```diff
@@ -1,2 +1,113 @@
 # activerecord-dynamic_timeout
-Dynamic Timeouts within ActiveRecord
+Dynamic query timeouts within ActiveRecord!
+
+## Installation
+
+Add this line to your application's Gemfile:
+```ruby
+gem 'activerecord-dynamic_timeout', require: 'active_record/dynamic_timeout/railtie'
+```
+
+### Non-Rails Applications
+If you are using this gem in a non-rails application, you will need run the initializer manually.
+```ruby
+require "active_record/dynamic_timeout"
+ActiveRecord::DynamicTimeout::Initializer.initialize!
+```
+
+## Usage
+    
+```ruby
+class MyModel < ActiveRecord::Base
+  # Model code...
+end
+
+MyModel.with_timeout(5) do
+  MyModel.find(1)
+end
```

Comment:
I think I was curious about how to show this when each adapter is very different...

For example, for Mysql db adapters, `SELECT SLEEP(2)` would work.
Postgres would be `SELECT PG_SLEEP(2)`.
SQLite, is much more difficult as I explained in the readme since it only is a timeout for table locking.